### PR TITLE
Avoid removing user labels

### PR DESCRIPTION
**Pull request by tiraboschi**
**What this PR does / why we need it**:
Reconcile managed labels on operands without
removing user added ones.
Add unit and func tests.

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:

```jira-ticket
https://issues.redhat.com/browse/CNV-36231
```

**Release note**:

```release-note
Avoid removing user labels
```

### CODE REVIEW

**Review by nunnatsa — COMMENTED** [review]
Add three nit comment inline.

File: pkg/util/labels.go [hunk 1]
```diff
@@ -4,13 +4,32 @@ import (
...
 		tgt.Labels[key] = val
 	}
 }
+
+// CompareLabels reports whether src labels are contained into tgt ones; extra labels on tgt are ignored.
+// It returns true if the src labels map is a subset of the tgt one.
+func CompareLabels(src, tgt metav1.Object) bool {
+	if src.GetLabels() == nil {
+		return true
+	}
```
[the remark this answers is not in the document]
Comment (tiraboschi) [reply pkg/util/labels.go:26]: done

File: tests/func-tests/golden_image_test.go [hunk 2]
```diff
@@ -120,20 +120,20 @@ var _ = Describe("golden image test", Label("data-import-cron"), Serial, Ordered
...
-			Eventually(func(g Gomega) map[string]string {
+			Eventually(func(g Gomega) string {
 				unstructured, err := cli.DynamicClient().Resource(isGVR).Namespace(imageNamespace).Get(ctx, expectedIS.Name, metav1.GetOptions{})
 				g.Expect(err).ShouldNot(HaveOccurred())
 				g.Expect(runtime.DefaultUnstructuredConverter.FromUnstructured(unstructured.Object, is)).To(Succeed())
 
-				return is.GetLabels()
-			}).WithTimeout(time.Second * 15).WithPolling(time.Millisecond * 100).ShouldNot(HaveKey("test-label"))
+				return is.GetLabels()["app.kubernetes.io/part-of"]
+			}).WithTimeout(time.Second * 15).WithPolling(time.Millisecond * 100).ShouldNot(Equal("wrong-value"))
```
[the remark this answers is not in the document]
Comment (tiraboschi) [reply tests/func-tests/golden_image_test.go]: done

**Review by nunnatsa — APPROVED** [review]
/approve

**Pull request merged**